src/newgrf.cpp

```diff
@@ -549,8 +548,7 @@ static void InitNewGRFFile(const GRFConfig &config)
 		return;
 	}
 
-	newfile = new GRFFile(config);
-	_grf_files.push_back(_cur.grffile = newfile);
+	_cur.grffile = &_grf_files.emplace_back(config);
```

Comment:
Maybe assert on the vector not growing here:
```suggestion
        assert(_grf_files.size() < _grf_files.capacity()); // we must not invalidate pointers
	_cur.grffile = &_grf_files.emplace_back(config);
```